processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java

```diff
@@ -74,7 +74,7 @@ public Sequence<T> run(final QueryPlus<T> input, final ResponseContext responseC
 
     final Thread currThread = setName ? Thread.currentThread() : null;
     final String currThreadName = setName ? currThread.getName() : null;
-    final String newName = setName ? query.getType() + "_" + query.getDataSource() + "_" + query.getIntervals() : null;
+    final String newName = setName ? "processing_" + query.getDataSource().getTableNames() + "_" + query.getId() : null;
```

Comment:
in case of multiple table names, do you want to join the set?

Reply:
Is there any particular reason why are we removing the query type and setting it to `processing_` now ?

Reply:
Honestly, I think it's fine to just set it to the Id and forget both the data source name and the query type.

Reply:
> Is there any particular reason why are we removing the query type and setting it to processing_ now ? 
So that stack traces reveal which queryID was blocking the execution pool. 

> Honestly, I think it's fine to just set it to the Id and forget both the data source name and the query type.

Thanks I reverted to my original commit.